src/core/tsi/ssl_transport_security_utils.cc

```diff
@@ -313,4 +313,50 @@ absl::StatusOr<std::string> IssuerFromCert(X509* cert) {
   return ret;
 }
 
+absl::StatusOr<std::string> AkidFromCertificate(X509* cert) {
+  if (cert == nullptr) {
+    return absl::InvalidArgumentError("cert cannot be null.");
+  }
+  ASN1_OCTET_STRING* akid = nullptr;
+  int j = X509_get_ext_by_NID(cert, NID_authority_key_identifier, -1);
+  // Can't have multiple occurrences
+  if (j >= 0) {
+    if (X509_get_ext_by_NID(cert, NID_authority_key_identifier, j) != -1) {
+      return absl::InvalidArgumentError("Could not get AKID from certificate.");
+    }
+    akid = X509_EXTENSION_get_data(X509_get_ext(cert, j));
+  }
```

Comment:
And if j is negative because we could not find an AKID in the certificate? 

I think you should check for this here and indicate what you intend to do in that case. 

Right now I think you are falling through calling i2d_ASN1_OCTET_STRING with a nullptr and then 
hitting the failure case of "Could not get AKID from certificate" - which is kind of ugly. 

If you intend to *require* an AKID in the certificate you should probably fail if j is negative here 
with a specific error that the certificate does not contain an AKID. Otherwise if you are going
to tolerate it not being there you need to do something else.

Reply:
Added an `else` block aligned with the `if j >= 0` statement that should catch and return a bad status if j is negative, good catch!